packages/zowe-explorer-api/src/tree/IZoweTree.ts

```diff
@@ -63,14 +64,27 @@ export interface IZoweTree<T> extends vscode.TreeDataProvider<T> {
      * Adds a session to the container
      * @param sessionName
      * @param type e.g. zosmf
+     * @param provider tree provider to add to, undefined will add for all
      */
-    addSession(sessionName?: string, type?: string): Promise<void>;
+    addSession(sessionName?: string, type?: string, provider?: IZoweTree<IZoweTreeNode>): Promise<void>;
+
+    /**
+     * Adds a single session to the tree
+     * @param profile the profile to add to the tree
+     */
+    addSingleSession(profile: imperative.IProfileLoaded): Promise<void>;
```

Comment:
should this be changed to an optional? (and maybe make it required in V3?)

Reply:
This function already existed in the definitions for trees such as `DataSetTree.ts`, `USSTree.ts` and `ZosJobsProvider.ts`. The problem was that it was never formally defined in the interface for access to this function from its parent class `ZoweTreeProvider.ts`.

A good example of this is installing the FTP extension and attempting to perform the `Add profile` action for all trees or singular tree and this should not show an issue

Another thing to note we reference this in our code but we don't really use an extender provided function, this is already part of our existing code but needs to be added to the interface for access